spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -304,4 +305,9 @@ public boolean caseSensitive() {
         .defaultValue(SQLConf.CASE_SENSITIVE().defaultValueString())
         .parse();
   }
+
+  public String branch() {
+    return confParser.stringConf().option(SparkWriteOptions.BRANCH).defaultValue(SnapshotRef.MAIN_BRANCH).parse();
```

Comment:
I think it would be nice to add a session config as well, which could be used to write to a branch in SQL. Then we could test DELETE FROM, MERGE INTO, UPDATE, and INSERT (OVERWRITE) queries as well.